staging/src/kubevirt.io/api/core/v1/types.go

```diff
@@ -1678,6 +1679,34 @@ type VirtualMachineStatus struct {
 	// RunStrategy tracks the last recorded RunStrategy used by the VM.
 	// This is needed to correctly process the next strategy (for now only the RerunOnFailure)
 	RunStrategy VirtualMachineRunStrategy `json:"runStrategy,omitempty" optional:"true"`
+
+	// VolumeMigrationState contains the information about the volumes set
+	// updates related to the volumeUpdateStrategy
+	VolumeUpdateState *VolumeUpdateState `json:"volumeUpdateState,omitempty" optional:"true"`
+}
+
+type VolumeUpdateState struct {
+	// VolumeMigrationState tracks the information related to the volume migration
+	VolumeMigrationState *VolumeMigrationState `json:"volumeMigrationState,omitempty" optional:"true"`
+}
+
+type VolumeMigrationState struct {
+	// MigratedVolumes lists the source and destination volumes during the volume migration
+	// +listType=atomic
+	// +optional
+	MigratedVolumes []StorageMigratedVolumeInfo `json:"migratedVolumes,omitempty"`
+	// The time the migration action began
+	// +nullable
+	StartTimestamp *metav1.Time `json:"startTimestamp,omitempty"`
+	// The time the migration action ended
+	// +nullable
+	EndTimestamp *metav1.Time `json:"endTimestamp,omitempty"`
+	// Indicates the migration completed
+	Completed *bool `json:"completed,omitempty"`
+	// Indicates that the migration succeeded
+	Succeeded *bool `json:"succeeded,omitempty"`
+	// Contains the reason why the migration failed
+	FailureReason string `json:"failureReason,omitempty"`
```

Comment:
I feel that migration Succeeded/Completed are great candidates for conditions. But if the feeling is that we are currently suffering from "condition overload" I am okay with keeping the info here. My only concern is that the fields will be redundant if we ever do add the conditions. Maybe not the worst thing in the world. But something to consider.  Thoughts @vladikr?

Reply:
I'm not sure why we need all of this data here.
Can't we simply keep the migration UUID - all of the timestamps and statuses will be written there. 

I thought we only needed an indication here of whether a manual recovery is needed.

Reply:
We could avoid the start and end timestamp , but the rest is quite essential imo.

Reply:
Can all this data really be summarized with a single bool? Then a condition would be the way to go I'd think. My only concern would be the lack of `MigratedVolumes` if user does not remember/save old volume name

Reply:
@mhenriks I mean keeping the migrated volumes plus a bool if it needs a recovery. Like:
```golang
type VolumeMigrationState struct {
	// MigratedVolumes lists the source and destination volumes during the volume migration
	MigratedVolumes []StorageMigratedVolumeInfo `json:"migratedVolumes,omitempty"`
        ManualRecovery *bool `json:"manualRecovery,omitempty"`
```

Reply:
I prefer the additional fields as currently defined in the PR but am okay this if you want to be super minimal. Would probably call the field `ManualRecoveryRequired` though

Reply:
The semantics are different, even if the implementation is quite similar for both cases.

The current version in the code, it expresses the result of the migration, so it doesn't explicitly says that the recovery is required. It is up to the user to understand and take action.

About the last proposed solution with the recovery field, it is quite similar to the restartrequired condition

Reply:
This is correct. VMIMs are linked to VMIs not, not VMs

Reply:
@xpivarc we can look into changing the behaviors and only remove the VMIMs when the VM is deleted (in case the VMI is lined to a VM).

Reply:
I like the way the status looks right now except for the `ManualRecovery`.
I tend to agree with @mhenriks about the `ManualRecoveryRequired`, but I don't have a strong opinion.
I misunderstood the reason why should we explicitly say whether manual recovery is required.

Reply:
@vladikr I'm not quite sure I understand which version you prefer, the one in the code or the minimal one with only the migrated volumes and the recovery boolean?

Reply:
Sorry that I wasn't clear. I prefer the [minimal version](https://github.com/kubevirt/kubevirt/pull/12355#discussion_r1750278829). 
The only open issue for me is the `ManualRecovery` field. I wonder why we can't be explicit about this. 
Could we say with confidence that Manual Recovery is required?

Reply:
@vladikr yes, because the volumes in vm spec are the new one but the copy wasn't successful or didn't finished

Reply:
How would you model this otherwise?

Reply:
I'm only referring to the filed name, s/ManualRecovery/ManualRecoveryRequired